workflows/codefresh-csdp/versions/1.1.3/images/report-image-info/src/registry-client.js

```diff
@@ -0,0 +1,169 @@
+                credentials: inputs.aws.credentials,
+            })
+        }
+        throw new Error('Registry credentials for ECR not set. Add following registry parameters in your workflow to continue:\n - AWS_ACCESS_KEY\n - AWS_SECRET_KEY\n - AWS_REGION\n');
+    }
+    if (checkNotEmpty(inputs.generic.credentials.username)
+        && checkNotEmpty(inputs.generic.credentials.password)
+        && checkNotEmpty(inputs.generic.request.host)) {
+        return new StandardRegistry(inputs.generic);
+    }
+    if (inputs.dockerConfigPath) {
+        return getCredentialsFromDockerConfig(image);
+    }
+    throw new Error('Registry credentials is required parameter. Add one from following registry parameters in your workflow to continue:\n - Docker credentials: DOCKERHUB_USERNAME, DOCKERHUB_PASSWORD\n - GCR credentials: GCR_KEY_FILE_PATH\n - AWS registry credentials: AWS_ACCESS_KEY, AWS_SECRET_KEY, AWS_REGION\n - Standard registry credentials: REGISTRY_USERNAME, REGISTRY_PASSWORD, REGISTRY_DOMAIN\n')
+}
+
+async function createRegistryClient(image) {
+    if (inputs.dockerConfigPath) {
+        return getCredentialsFromDockerConfig(image)
+    }
+
+    // Clean this up when have time
+    if (checkNotEmpty(inputs.dockerhub.username)
+        && checkNotEmpty(inputs.dockerhub.password)) {
+
+        return new DockerhubRegistry(inputs.dockerhub);
+    }
+
+    if (checkNotEmpty(inputs.generic.credentials.username)
+        && checkNotEmpty(inputs.generic.credentials.password)
+        && checkNotEmpty(inputs.generic.request.host)) {
+
+        return new StandardRegistry(inputs.generic);
+    }
+
+    if (checkNotEmpty(inputs.aws.role)
+        && checkNotEmpty(inputs.aws.credentials.region)) {
+        return createECRUsingSTS(inputs.aws.role, inputs.aws.credentials.region);
+    }
+
+    if (inputs.gcr.keyFilePath) {
+        return new GcrRegistry({
+            keyfile: fs.readFileSync(inputs.gcr.keyFilePath),
+            request: { host: 'gcr.io' }
+        });
+    }
+    if (checkNotEmpty(inputs.aws.credentials.accessKeyId)
+        && checkNotEmpty(inputs.aws.credentials.secretAccessKey)
+        && checkNotEmpty(inputs.aws.credentials.region)) {
+        return new EcrRegistry({
+            promise: Promise,
+            credentials: inputs.aws.credentials,
+        })
+    }
+    throw new Error('Registry credentials is required parameter. Add one from following registry parameters in your workflow to continue:\n - Docker credentials: DOCKERHUB_USERNAME, DOCKERHUB_PASSWORD\n - GCR credentials: GCR_KEY_FILE_PATH\n - AWS registry credentials: AWS_ACCESS_KEY, AWS_SECRET_KEY, AWS_REGION\n - Standard registry credentials: REGISTRY_USERNAME, REGISTRY_PASSWORD, REGISTRY_DOMAIN');
+}
+
+module.exports = {
+    parseImageName,
+    async getRegistryClient(image) {
+    if (inputs.retrieveCredentialsByDomain) {
+        return createRegistryClientByImage(image);
+    }
+    return createRegistryClient(image);
```

Comment:
please extract the function and bind if you need `this` instance or just export if not

Reply:
done